server/utils/koa.utils.ts

```diff
@@ -15,6 +15,19 @@ import { MODE, viteDevServer } from '../index.js';
 
 export const setupCompression = (app: Koa) => app.use(compress());
 
+export const setupHTTPSRedirect = (app: Koa) => {
+  app.use(async (ctx, next) => {
+    const { request } = ctx;
+    const proto = request.protocol;
+
+    if (proto === 'http') {
+      ctx.redirect(`https://${request.host}${request.url}`);
+    } else {
+      await next();
+    }
+  });
+};
+
```

Comment:
```suggestion
export const setupHTTPSRedirect = (app: Koa) => {
  app.use(async (ctx, next) => {
    const { protocol, host, url } = ctx.request;

    if (protocol === 'http' && !host.includes('localhost')) {
      ctx.redirect(`https://${host}${url}`);
    } else {
      await next();
    }
  });
};

```
i would suggest to consider allowing http in combination with localhost so you can still propperly build and start tha app locally

i see a little bit of risk in checking `!host.includes('localhost')` beacause it could be the case that a production build is also running on localhost behind a loadbalancer or whatever. maybe we could also consider using a ENV to achieve this so in local development you can just switch this variable to false